pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -448,17 +448,20 @@ func (s *DiscoveryServer) WorkloadUpdate(id string, workloadLabels map[string]st
 		adsClientsMutex.RLock()
 		for _, connection := range adsClients {
 			// if the workload has envoy proxy and connected to server,
-			// then do a full xDS push for this proxy;
+			// then do a full xDS push for this proxy,
+			// need to consider multi-ip scenarios, such as cni or multus-cni;
 			// otherwise:
 			//   case 1: the workload has no sidecar proxy, no need xDS push at all.
 			//   case 2: the workload xDS connection has not been established,
 			//           also no need to trigger a full push here.
-			if connection.modelNode.IPAddresses[0] == id {
-				// There is a possibility that the pod comes up later than endpoint.
-				// So no endpoints add/update events after this, we should request
-				// full push immediately to speed up sidecar startup.
-				s.pushQueue.Enqueue(connection, &model.PushRequest{Full: true, Push: s.globalPushContext(), Start: time.Now()})
-				break
+			for _, ip := range connection.modelNode.IPAddresses {
```

Comment:
I dont think you should modify here, as only the first ip (podip) is used to build node id.

Reply:
OK, I found there is a mistake  I will modify soon